doc/tools/make_rst.py

```diff
@@ -1261,7 +1299,9 @@ def make_method_signature(
     return ret_type, out
 
 
-def make_heading(title, underline):  # type: (str, str) -> str
+def make_heading(title, underline, l10n=True):  # type: (str, str, bool) -> str
+    if l10n:
+        title = headings_l10n.get(title, title)
     return title + "\n" + (underline * len(title)) + "\n\n"
```

Comment:
reStructuredText requires the underlines' length to be visually the same as the title at least.

While after l10n, `title` may contain wide characters, so it should be `len(title) * 2` here.

Reply:
Hm, but Sphinx also throws warnings if the underlines' length is not *exactly* the same as the text :|
Can we figure out what the exact correct length would be?

Reply:
Might be worth looking up what test Sphinx itself performs to see if we can replicate it. Alternatively, maybe there's a lint hint that we can set in generated files to ignore the warning.

Reply:
Ah, it uses something like this: https://github.com/docutils-mirror/docutils/blob/e88c5fb08d5cdfa8b4ac1020dd6f7177778d5990/docutils/utils/__init__.py#L618-L642

Unicode combining characters are also taken into account 😲

Reply:
Yeah I was about to link that :D

But I also found that I'm mistaken, it doesn't complain if the underline is too long, only if it's too short. So your solution to multiply by 2 is fine :+1:

Reply:
Should be fixed now.